thor/params.go

```diff
@@ -65,3 +65,21 @@ var (
 	EnergyGrowthRate      = big.NewInt(5000000000) // WEI THOR per token(VET) per second. about 0.000432 THOR per token per day.
 	NumberOfBlocksPerYear = big.NewInt(8640 * 365) // number of blocks per year, non leap (365 days)
 )
+
+// GetMaxBlockProposers retrieves the max block proposers parameter with fallback to initial value
+// If capToInitial is true, values greater than InitialMaxBlockProposers are also capped (PoA)
+func GetMaxBlockProposers(params interface {
```

Comment:
Do you think this function could receive the default max block proposers value instead of the first interface? This way it only holds the following logic that could change the value based on the `capToInitial`, which is the duplicated one. Also, if the `cappedToInitial` is something bounded to `PoA`, we could rename it something more explicit, like `isPoA`

Reply:
I think it is better this way. Otherwise we would need to repeat this:

```
       // Where params can be different things
	mbp, err := params.Get(KeyMaxBlockProposers)
	if err != nil {
		return 0, err
	}
	
	// And now thor.GetMaxProposers(mbp)
```

With the current approach is a one-liner.